src/ngraph/op/quantized_convolution.cpp

```diff
@@ -0,0 +1,196 @@
+    , m_window_movement_strides(window_movement_strides)
+    , m_window_dilation_strides(window_dilation_strides)
+    , m_padding_below(padding_below)
+    , m_padding_above(padding_above)
+    , m_data_dilation_strides(data_dilation_strides)
+    , m_output_type(output_type)
+    , m_input_axes(input_axes)
+    , m_filter_axes(filter_axes)
+    , m_output_axes(output_axes)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::QuantizedConvolution::validate_and_infer_types()
+{
+    enum
+    {
+        INPUT,
+        FILTER,
+        INPUT_SCALE,
+        INPUT_ZERO_POINT,
+        FILTER_SCALE,
+        FILTER_ZERO_POINT,
+        OUTPUT_SCALE,
+        OUTPUT_ZERO_POINT
+    };
+
+    NODE_VALIDATION_CHECK(
+        this, m_output_type.is_static(), "Output element type must not be dynamic");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_output_type.is_quantized(),
+                          "Output element type (",
+                          m_output_type,
+                          ") must be a quantized type");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(INPUT).is_quantized(),
+                          "Input element type (",
+                          get_input_element_type(INPUT),
+                          ") must be a quantized type");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(FILTER).is_quantized(),
+                          "Filter element type (",
+                          get_input_element_type(FILTER),
+                          ") must be a quantized type");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(INPUT_SCALE).is_real() ||
+                              get_input_element_type(INPUT_SCALE).is_dynamic() ||
+                              get_input_element_type(FILTER_SCALE).is_real() ||
+                              get_input_element_type(FILTER_SCALE).is_dynamic() ||
+                              get_input_element_type(OUTPUT_SCALE).is_real() ||
+                              get_input_element_type(OUTPUT_SCALE).is_dynamic(),
+                          "Scale must be a floating point number");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_element_type(0).compatible(get_input_element_type(INPUT_ZERO_POINT)),
+        "Input Zero point element type (",
+        get_input_element_type(INPUT_ZERO_POINT),
+        ") must match input element type (",
+        get_input_element_type(0),
+        ")");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_element_type(1).compatible(get_input_element_type(FILTER_ZERO_POINT)),
+        "Filter Zero point element type (",
+        get_input_element_type(FILTER_ZERO_POINT),
+        ") must match filter element type (",
+        get_input_element_type(1),
+        ")");
+
+    // TODO Remove these checks once we support channelwise and vector of scales
```

Comment:
mxnet bridge already uses vector of scales. Why are we breaking compatibility with that? We could let the op definition support a vector and its ok for reference kernel to not support it for now.